### PR TITLE
vectorstores: fix weaviate not return score from SimilaritySearch

### PR DESCRIPTION
**Pull request by Abirdcfly**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: vectorstores/weaviate/weaviate.go [hunk 1]
```diff
@@ -197,10 +201,19 @@ func (s Store) parseDocumentsByGraphQLResponse(res *models.GraphQLResponse) ([]s
 		if !ok {
 			return nil, ErrMissingTextKey
 		}
+		additional, ok := itemMap["_additional"].(map[string]any)
```
Comment (tmc) [inline vectorstores/weaviate/weaviate.go:204]: what are the conditions/guarantees that this is present?

Comment (tmc) [reply vectorstores/weaviate/weaviate.go:204]: if it isn't always present it seems like we should return the document without a populated score vs return an error earlier, thoughts?

Comment (Abirdcfly) [reply vectorstores/weaviate/weaviate.go:204]: > what are the conditions/guarantees that this is present?
`certainty` in `_additional` is weaviate's built-in attributes:
https://weaviate.io/developers/weaviate/api/graphql/additional-properties

and we add this in https://github.com/tmc/langchaingo/blob/e98eb1703a8d377ef3efeaa68fb24b33ef34e726/vectorstores/weaviate/weaviate.go#L262

but I agress we can return the document without a populated score.

**Review by tmc — APPROVED** [review]
LGTM

**Pull request merged**